m2m-transfers/V3/REPWRITERSPECS/BANNO.M2MTRANSFERS.RPT.BATCH

```diff
@@ -0,0 +1,751 @@
+         IF LENGTH(BNOTEXT)>0 THEN
+          CALL BNONLTS
+         TARGETCOMMENT=BNOTEXT
+        END          
+       IF UPPERCASE(SEGMENT(CFGFILELINE,1,3))="XA:" THEN
+        DO
+         BNOTEXT=SEGMENT(CFGFILELINE,4,LENGTH(CFGFILELINE))
+         IF LENGTH(BNOTEXT)>0 THEN
+          CALL BNONLTS
+         IF CHARACTERSEARCH("YESTRUEON",UPPERCASE(BNOTEXT))>0 THEN
+          XACCTACCESS=TRUE
+        END  
+      END [IF CFGFILEERROR=""...]
+     ELSE IF CFGFILEERROR<>"EOF" THEN
+      DO
+       PRINT "Error reading Letter file '"+CFGFILENAME+"': "+CFGFILEERROR
+       NEWLINE
+       PRINT "TERMINATING"
+       NEWLINE
+       TERMINATE
+      END
+    END [WHILE CFGFILEERROR=""]
+  END [IF CFGFILEERROR=""]
+ ELSE
+  DO
+   PRINT "Letter file '"+CFGFILENAME+"' open error:"+CFGFILEERROR
+   NEWLINE
+   PRINT "TERMINATING"
+   NEWLINE
+   TERMINATE
+  END
+
+[* Print parameters for this run to the Batch Output report
+*]
+ PRINT FORMAT("TARGETSYMXUSER:9999",TARGETSYMXUSER)
+ NEWLINE
+ PRINT FORMAT("TARGETSTARTDATE:99/99/99",TARGETSTARTDATE)
+ NEWLINE
+ PRINT FORMAT("TARGETENDDATE:99/99/99",TARGETENDDATE)
+ NEWLINE
+ PRINT "TARGETCOMMENT:'"+TARGETCOMMENT+"'"
+ NEWLINE
+ PRINT "XACCTACCESS: "+BNOTRUEFALSECHR(XACCTACCESS)
+ NEWLINE
+ NEWLINE
+
+ IF TARGETCOMMENT="" THEN
+  DO
+   PRINT "Target Comment not determined - TERMINATING PROGRAM RUN."
+   NEWLINE
+   TERMINATE
+  END
+END [PROCEDURE]
+
+PROCEDURE GETTRANSACTIONS
+[* Target SymX transactions (by user number)
+** Load a transaction sequence:
+**   A transaction followed by one or more comments
+** Test the transaction:
+**   Either the second comment after the transaction will have the
+**   TARGETCOMMENT as part of the transaction comment field, OR
+**   the TARGETCOMMENT will be part of the transaction
+**   statement description 1 field.
+*]
+ FOR EACH SHARE WITH (SHARE:CLOSEDATE=DATENULL OR
+                      SHARE:CLOSEDATE>=TARGETSTARTDATE)
+  DO
+   TRANSOURCE=TRANSOURCESHARE
+   TRANSOURCEID=SHARE:ID
+   FOR EACH SHARE TRANSACTION WITH (SHARE TRANSACTION:ACCOUNTNUMBER=ACCOUNT:NUMBER AND
+                                    SHARE TRANSACTION:POSTDATE>=TARGETSTARTDATE AND
+                                   (SHARE TRANSACTION:POSTDATE<=TARGETENDDATE OR
+                                    TARGETENDDATE=DATENULL) AND
+                                    SHARE TRANSACTION:VOIDCODE=0 AND
+                                    SHARE TRANSACTION:USERNUMBER=TARGETSYMXUSER)
+    DO
+     IF SHARE TRANSACTION:COMMENTCODE=1 THEN
+      DO
+       IF TCCOUNT<TCCOUNTMAX THEN [store pertinant data for future use]
+        DO
+         TCCOUNT=TCCOUNT+1
+         TCCOMMENT(TCCOUNT)=SHARE TRANSACTION:COMMENT
+         TCIDTYPE(TCCOUNT)=SHARE TRANSACTION:IDTYPE
+         TCID(TCCOUNT)=SHARE TRANSACTION:ID
+         TCSTMTDESCR(TCCOUNT)=SHARE TRANSACTION:STMTDESC:1
+        END
+      END
+     ELSE [transaction segment is not a comment - validate & process]
+      DO
```

Comment:
I think this is right that the check doesn't happen until their is at least 1 transaction comment. Is that correct that the M2M Code always makes at least one comment transaction as well when making an instant transfer?

Reply:
@jordan-keenan My testing showed that there would be at least 1 comment record after the transaction and 2 comment records IF the comment was over 19 characters (I think)!